tests/sentry/integrations/slack/test_disable.py

```diff
@@ -0,0 +1,163 @@
+from sentry.testutils.helpers import install_slack, with_feature
+from sentry.testutils.silo import region_silo_test
+from sentry.utils import json
+
+control_address = "http://controlserver"
+secret = "hush-hush-im-invisible"
+
+
+@override_settings(
+    SENTRY_SUBNET_SECRET=secret,
+    SENTRY_CONTROL_ADDRESS=control_address,
+)
+class SlackClientDisable(TestCase):
+    def setUp(self):
+        self.resp = responses.mock
+        self.resp.__enter__()
+
+        self.integration = Integration.objects.create(
+            provider="slack",
+            name="Awesome Team",
+            external_id="TXXXXXXX1",
+            metadata={
+                "access_token": "xoxb-xxxxxxxxx-xxxxxxxxxx-xxxxxxxxxxxx",
+                "installation_type": "born_as_bot",
+            },
+        )
+        self.integration.add_organization(self.event.project.organization, self.user)
+        self.payload = {"channel": "#announcements", "message": "i'm ooo next week"}
+
+    def tearDown(self):
+        self.resp.__exit__(None, None, None)
+
+    @responses.activate
+    @with_feature("organizations:disable-on-broken")
+    def test_fatal_and_disable_integration(self):
+        bodydict = {"ok": False, "error": "account_inactive"}
+        self.resp.add(
+            method=responses.POST,
+            url="https://slack.com/api/chat.postMessage",
+            status=200,
+            content_type="application/json",
+            body=json.dumps(bodydict),
+        )
+        client = SlackClient(integration_id=self.integration.id)
+        with pytest.raises(ApiError):
+            client.post("/chat.postMessage", data=self.payload)
+        buffer = IntegrationRequestBuffer(client._get_redis_key())
+        assert buffer.is_integration_broken() is True
+        assert (
+            integration_service.get_integration(integration_id=self.integration.id).status
+            == ObjectStatus.DISABLED
+        )
+
+    @responses.activate
+    def test_fatal_integration(self):
+        """
+        fatal fast shut off with flag off
+        """
+        bodydict = {"ok": False, "error": "account_inactive"}
+        self.resp.add(
+            method=responses.POST,
+            url="https://slack.com/api/chat.postMessage",
+            status=200,
+            content_type="application/json",
+            body=json.dumps(bodydict),
+        )
+        client = SlackClient(integration_id=self.integration.id)
+        with pytest.raises(ApiError):
+            client.post("/chat.postMessage", data=self.payload)
+        buffer = IntegrationRequestBuffer(client._get_redis_key())
+        assert buffer.is_integration_broken() is True
+        assert (
+            integration_service.get_integration(integration_id=self.integration.id).status
+            == ObjectStatus.ACTIVE
+        )
+
+    @responses.activate
+    def test_error_integration(self):
+        bodydict = {"ok": False, "error": "The requested resource does not exist"}
```

Comment:
Can you add a docstring here explaining that this test covers the case when we encounter an error that indicates an integration is broken we log it, but do not yet disable it because it hasn't happened x number of times. Try to think of docstrings (especially test ones) from the perspective of someone who looks at this w/o much context - sometimes it's future you also 😛